contrib/kafka-container-source/cmd/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/Shopify/sarama"
+	cluster "github.com/bsm/sarama-cluster"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/google/uuid"
+)
+
+var (
+	sink      string
+	topic     string
+	groupID   string
+	bootstrap string
+)
+
+func init() {
+	flag.StringVar(&sink, "sink", "", "the host url to send messages to")
+	flag.StringVar(&topic, "topic", "", "the topic to read messages from")
+	flag.StringVar(&groupID, "groupId", uuid.New().String(), "the consumer group")
+	flag.StringVar(&bootstrap, "bootstrap", "localhost:9092", "Apache Kafka bootstrap servers")
+}
+
+func main() {
+
+	flag.Parse()
+
+	kafkaConfig := cluster.NewConfig()
+	kafkaConfig.Consumer.Return.Errors = true
+	kafkaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	kafkaConfig.Consumer.Offsets.CommitInterval = 10 * time.Second
+	kafkaConfig.Group.Return.Notifications = true
+
+	ce, err := client.NewHTTPClient(
+		client.WithTarget(sink),
+		client.WithHTTPBinaryEncoding(),
+		client.WithUUIDs(),
+		client.WithTimeNow(),
+	)
+	if err != nil {
+		log.Fatalf("Failed to create a http cloudevent client: %s", err.Error())
+	}
+
+	// init consumer
+	brokers := []string{bootstrap}
+	topics := []string{topic}
+
+	consumer, err := cluster.NewConsumer(brokers, groupID, topics, kafkaConfig)
+	if err != nil {
+		panic(err)
```

Comment:
I'm wondering about the choice of log.Fatal() above and a panic() here, how are these two cases different?